dfget/core/api/api_test.go

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 1999-2018 Alibaba Group.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package api
+
+import (
+	"testing"
+	"time"
+
+	"github.com/go-check/check"
+)
+
+func Test(t *testing.T) {
+	check.TestingT(t)
```

Comment:
Is this the suggested approach for unit testing in this project?

Reply:
Yes, go-check provides more handy functions for unit testing.

Reply:
I'm not sure how much adoption that go-check has. I always find that simple table driven unit tests are enough.

Reply:
All of them are allowed to write unit-test in this project. IMOP, `go-check` is much more handy and clear than `Golang testing` in some scenarios, such as `util_test.go`, `rate_limiter_test.go`. And I didn't like to write `if statement` to check whether the result is expected in unit testing.